tests/integration/services/test_request.py

```diff
@@ -0,0 +1,71 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2024 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Integration tests for the Request service."""
+import hashlib
+from unittest.mock import call
+
+import pytest
+import pytest_check
+
+
+@pytest.mark.parametrize(
+    ("url", "checksum", "size"),
+    [
+        pytest.param(
+            # A real build log.
+            "https://launchpad.net/~charmcraft-team/charmcraft/+snap/charmcraft-edge/+build/2377699/+files/buildlog_snap_ubuntu_jammy_armhf_charmcraft-edge_BUILDING.txt.gz",
```

Comment:
Do we know if this file is going to stay there forever?

Reply:
As far as I can tell public build logs are available indefinitely, but it's possible that if we delete that recipe from launchpad it'll disappear.